### ISSUE TITLE
Question: Memory mapped file protection settings READ_ONLY vs READ_WRITE

### ISSUE THREAD

**Issue by abellina** [openucx/ucx#4924]
I am trying to figure out why the following seems required.

For a memory mapped file, protection settings (`prot` argument in `mmap`: http://man7.org/linux/man-pages/man2/mmap.2.html) seem to be required to be `READ_WRITE`, rather than `READ_ONLY` while registering memory in the `rc` TLS.

Is this expected?

Note: the failure is:

```
[1584981156.953034] [dgx2h0194:33577:1]          ib_md.c:330  UCX  ERROR ibv_exp_reg_mr(address=0x7fb7c609e000, length=2630912, access=0xf) failed: Cannot allocate memory
[1584981157.294035] [dgx2h0194:33577:1]         rcache.c:591  UCX  DEBUG failed to register region 0x7fab9c466b10 [0x7fb7c609e000..0x7fb7c6320500]: Input/output error
[1584981157.294043] [dgx2h0194:33577:1]         ucp_mm.c:137  UCX  ERROR failed to register address 0x7fb7c609e000 mem_type bit 0x1 length 2630912 on md[2]=mlx5_0: Input/output error (md reg_mem_types 0x17)
```

**Comment by yosefe** [comment]
@abellina where do you see it's READ_ONLY for rc ?

**Comment by yosefe** [comment]
do you mean that UCX fails to register a read-only memory mapping?

**Comment by abellina** [comment]
yes that's what I mean @yosefe. sparkucx is also using READ_WRITE, if I read this correctly:

https://github.com/openucx/sparkucx/blob/9b6d5185db0dd44639332a2afbad360122aa5ff3/src/main/java/org/apache/spark/shuffle/ucx/UnsafeUtils.java#L48

Before this I was trying with `READ_ONLY` and saw the errors, and noticed that switching to `READ_WRITE` made it work.

**Comment by Akshay-Venkatesh** [comment]
@yosefe Want to check if this problem been resolved. We have another user who's run into this issue but we've not confirmed yet if the buffer being passed is mmap'ed with `READ_ONLY` flags.

**Comment by yosefe** [comment]
@Akshay-Venkatesh it hasn't been resolved yet
